src/components/Camera/Video.js

```diff
@@ -1,64 +1,187 @@
     }
   }
 
-  buttonClass = () => classNames({[style.stopRecording]: this.state.recording, [style.startRecording]: !this.state.recording })
+  handleTimeout = () => {
+    this.setState({ hasTimedOut: true })
+  }
+
+  handleRedoVideo = () => {
+    this.loadChallenges()
+  }
+
+  handleNextChallenge = () => {
+    this.setState({ currentIndex: this.state.currentIndex + 1 })
+  }
+
+  renderRedoAction = () => (
+    <button
+      onClick={preventDefaultOnClick(this.handleRedoVideo)}
+      className={classNames(theme.btn, theme['btn-ghost'])}
+    >{this.props.i18n.t('capture.liveness.challenges.redo_video')}</button>
+  )
+
+  renderChallenges = () => {
+    const { i18n } = this.props
+    const { isRecording, hasTimedOut, currentIndex, challenges = [] } = this.state
+    const { type, value } = challenges[currentIndex] || {}
+    const isLastChallenge = currentIndex === challenges.length - 1
 
-  render() {
     return (
-      <div>
-        <CameraPure {...{...this.props, video: true, webcamRef: (c) => { this.webcam = c }}}/>
-        <CaptureActions {...this.props}
-          btnText={this.buttonText()}
-          handleClick={this.handleVideoClick}
-          btnClass={this.buttonClass()}
-        />
+      <div className={style.livenessCamera}>
+        <CameraPure {...{
+          ...this.props,
+          video: true,
+          isFullScreen: true,
+          isLiveness: true,
+          webcamRef: c => this.webcam = c,
+          title: '',
+          subTitle: '',
+          ...(hasTimedOut ? { error: timeoutError } : {})
+        }} />
+        <div className={style.caption}>
+        {
+          isRecording ?
+            <div>
+              <div className={style.recordingIndicator}>
+                {i18n.t('capture.liveness.recording')}
+              </div>
+              <Challenge {...{i18n, type, value}} />
+            </div> :
+            i18n.t('capture.face.webcam')
+        }
+        </div>
+        <CaptureActions
+          hint={ isRecording ? i18n.t(`capture.liveness.challenges.done_${ isLastChallenge ? 'stop' : 'next' }`) : '' }
+        >
+          {
+            !isLastChallenge && isRecording ?
+              <button
+                className={`${theme.btn} ${theme['btn-centered']} ${theme['btn-primary']}`}
+                onClick={this.handleNextChallenge}>
+                {this.props.i18n.t('capture.liveness.challenges.next')}
+              </button> :
+              <button
+                className={classNames(style.btn, {
+                  [style.stopRecording]: isRecording,
+                  [style.startRecording]: !isRecording,
+                })}
+                onClick={isRecording ? this.handleRecordingStop : this.handleRecordingStart} 
+              />
+          }
+        </CaptureActions>
       </div>
     )
   }
+
+  render() {
+    const { i18n } = this.props
+    const { hasSeenIntro, hasLoaded, hasError } = this.state
+
+    return (
+      hasSeenIntro ?
```

Comment:
Is the user supposed to see the intro again if they navigate back?

Reply:
I assumed they don't. @seewah?